Dockerfile.integration

```diff
@@ -0,0 +1,74 @@
+# Dockerfile.integration - For running integration tests with envtest
+# This Dockerfile sets up a complete environment for running k8s-client integration tests
+
+FROM golang:1.24-alpine AS builder
+
+# Install required tools
+RUN apk add --no-cache \
+    git \
+    make \
+    bash \
+    curl \
+    tar \
+    gzip
+
+# Set working directory
+WORKDIR /workspace
+
+# Copy go mod files first for better caching
+COPY go.mod go.sum ./
+RUN go mod download
+
+# Install setup-envtest tool
+RUN go install sigs.k8s.io/controller-runtime/tools/setup-envtest@latest
+
+# Download and setup envtest binaries (Kubernetes API server components)
+# This downloads etcd and kube-apiserver for testing
+RUN setup-envtest use 1.31.x --bin-dir /usr/local/kubebuilder/bin
+
+# Verify binaries are installed
+RUN ls -la /usr/local/kubebuilder/bin/ && \
+    /usr/local/kubebuilder/bin/kube-apiserver --version
```

Comment:
It is better to put line 27 & line 30 above `COPY go.mod go.sum ./` for this part is not related to code, so any update in go.mod will not break this cache.

And I also advice combine these two lines into one to reduce layers.